### PR TITLE
fix(build): switch to esbuild in place of pika

### PR DESCRIPTION
**Pull request by kfcampbell**
Replicating [octokit/request.js#584](https://github.com/octokit/request.js/pull/584) to another repo. 

---

Pika has been deprecated for a while now, and the project has now been archived, and because of that the released npm package is missing most of the files generated by the build step.

Uses esbuild to transpile the TS source code into an ESM source, NodeJs bundle, and a browser bundle

Uses the TypeScript compiler to generate the types.

### Before the change?

* Uses pika for the build system
* The released npm package is missing most of the files generated by the build step. `dist-node`, `dist-types`, `dist-web`... even though they are generated correctly.

### After the change?

* Uses esbuild to generate the JS bundles
* Uses the TypeScript compiler to generate the type definitions
* The correct file matching patterns are used so we publish all the necessary files again.

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/main/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:

* N/A

### Pull request type

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Comment by kfcampbell** [comment]
Good catch. Done!

**Comment by kfcampbell** [comment]
I imagine the pika CDN references can stay, correct?

**Comment by wolfy1339** [comment]
Yep those references can stay, for now.

File: package.json [hunk 1]
```diff
@@ -44,6 +42,14 @@
   },
   "jest": {
     "preset": "ts-jest",
```
Comment (wolfy1339) [inline package.json:44]:
```suggestion
```

File: tsconfig.json [hunk 2]
```diff
@@ -1 +1,11 @@
...
+  "extends": "@octokit/tsconfig",
+  "compilerOptions": {
+    "esModuleInterop": true,
+    "declaration": true,
+    "outDir": "pkg/dist-types",
+    "emitDeclarationOnly": true,
+    "sourceMap": true
+  },
+  "include": ["src/**/*"]
+}
```
Comment (wolfy1339) [inline tsconfig.json:11]:
```suggestion
}

```

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**